TACC/Core-Portal
# Task/FP-630 - Notify user to push keys before starting app

## Overview: ##
If the default storage system needs to have keys pushed, display a message to the user and lock the app form. The message should bring up the push keys modal, and should only unlock the form upon successful result.

## Related Jira tickets: ##

* [FP-630](https://jira.tacc.utexas.edu/browse/FP-630)

## Summary of Changes: ##

- Added the push keys modal and message to the top of the app form
- Return storage system status with app information (so we can determine if we need to push the keys)
- Added test to check for push keys message appearance
- Updated NAMD app fixture with additional data coming back from the AppView
- Created "reloadApp" function for the apps saga which will update the app state (regardless if it's the same app ID being returned)

## Testing Steps: ##
1. ssh into your default storage system and delete the authorized_keys there
2. navigate to the applications area and select an app. If your keys are missing (for the default storage system) you should see a message requiring you to push your keys.

## UI Photos:
![image](https://user-images.githubusercontent.com/29575979/110180944-62409000-7dd0-11eb-973e-bf796f4b41ca.png)
